### PR TITLE
[CALCITE-3592] Implement BITNOT scalar function

### PR DESCRIPTION
**Pull request by normanj-bitquill**
* Implemented the BITNOT scalar function as a standard function
* Supports integer and binary arguments
* Returns NULL if the argument is NULL

### CODE REVIEW

File: testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java [hunk 1]
```diff
@@ -15738,6 +15738,29 @@ private static void checkLogicalOrFunc(SqlOperatorFixture f) {
         true);
   }
 
+  @Test void testBitNotScalarFunc() {
+    final SqlOperatorFixture f = fixture();
+    f.setFor(SqlStdOperatorTable.BITXOR, VmName.EXPAND);
```
Comment (mihaibudiu) [inline testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:15743]: XOR?
How about a failing test with DECIMAL?

**Review by mihaibudiu — APPROVED** [review]

File: testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java (hunk 1 above)
Comment (normanj-bitquill) [reply testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java]: Interesting that using XOR had not impact on the tests. Fixed now.

Also added a test for `BITNOT(DECIMAL)`.

Comment (mihaibudiu) [reply testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java]: I think that setFor is just decorative

File: testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java [hunk 2]
```diff
@@ -15738,6 +15738,35 @@ private static void checkLogicalOrFunc(SqlOperatorFixture f) {
         true);
   }
 
+  @Test void testBitNotScalarFunc() {
```
Comment (caicancai) [inline testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:15741]: Add jira link

Comment (normanj-bitquill) [reply testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:15745]: Updated.

**Review by caicancai — APPROVED** [review]

**Comment by normanj-bitquill** [comment]
The commits have been squashed.

**Pull request merged**